src/fprime_extras/lint/rules/rule_valid_port_types.py

```diff
@@ -0,0 +1,162 @@
+                                            rule_slug=slug,
+                                            description=msg,
+                                            line_number=port.sourceline,
+                                            column=0,
+                                            severity=LintSeverity.CRITICAL
+                                            )
+                        notifications.append(note)
+        return notifications
+
+    def fix(self):
+        # if args.fix and flag:
+        #     conn.getparent().remove(conn)
+        pass
+
+    def reset(self):
+        pass
+
+    def get_imported_types(self, import_list, fprime_path, extras_file):
+        """
+        import_list: list of import_port_type etree elements
+        fprime_path: path to the fprime root
+        extras_file: file that imports the given files
+
+        parses the imported files and returns a hash set of their full type names as strings.
+        Will also return any critical notifications associated with parsing the imported types
+        """
+        imported_types = set()
+        notifications = []
+        for imp in import_list:
+            import_path = os.path.join(fprime_path, imp.text)
+            try:
+                xml_tree = etree.parse(import_path)
+                root = xml_tree.getroot()
+                if root.tag == "interface":
+                    name = root.get("name")
+                    namespace = root.get("namespace")
+                    full_name = namespace + "::" + name
+                    imported_types.add(full_name)
+                    log.info("Found port type: {}".format(full_name))
+                else:
+                    note = Notification(filename=extras_file.full_filename,
+                                        rule_slug=slug,
+                                        description="Component imported invalid port type: {}".format(imp.text),
+                                        line_number=imp.sourceline,
+                                        column=0,
+                                        severity=LintSeverity.CRITICAL
+                                        )
+                    notifications.append(note)
+            except XMLSyntaxError as e:
+                note = Notification(filename=extras_file.full_filename,
+                                    rule_slug=slug,
+                                    description=e.msg,
+                                    line_number=e.position[0],
+                                    column=e.position[1],
+                                    severity=LintSeverity.CRITICAL
+                                    )
+                notifications.append(note)
+            except OSError as e:
+                note = Notification(filename=extras_file.full_filename,
+                                    rule_slug=slug,
+                                    description=e,
```

Comment:
What happens when you see this exception?  I don't think we want to expose raw Python exception text as part of an error unless it describes exactly what the lint user needs to know.

Reply:
This is example output for this issue.
`**************** LedBlinker/LedBlinkerComponentAi.xml`
`[FP-LINT] (L5,C0) invalid-port-type-import -> Error reading file '../../Svc/Sched/SchedPortAbi.xml': failed to load external entity "../../Svc/Sched/SchedPortAbi.xml" [CRITICAL]`
I've only seen this with invalid file paths that can't be opened. I can use a check that the file exists to trigger this kind of message instead, but I wasn't sure if there might be other errors that would come up

Reply:
This is example output for this issue.
`**************** LedBlinker/LedBlinkerComponentAi.xml`
`[FP-LINT] (L5,C0) invalid-port-type-import -> Error reading file '../../Svc/Sched/SchedPortAbi.xml': failed to load external entity "../../Svc/Sched/SchedPortAbi.xml" [CRITICAL]`
I've only seen this with invalid file paths that can't be opened. I can use a check that the file exists to trigger this kind of message instead, but I wasn't sure if there might be other errors that would come up

Reply:
OK, I think we want to catch the most narrow exception here.  Other exceptions might have other meanings and contexts, which could indicate bugs.  If we don't know they will be lint, we shouldn't catch them.  While this is not true today, the hook-calling mechanism will eventually trap uncaught exceptions.  Then the user can be prompted to file a bug report, even if that bug is to emit a new lint condition.

Reply:
I switched to using a check that the imported file exists, instead of catching the exception